sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/DocumentServiceLeaseUpdaterImpl.java

```diff
@@ -90,10 +91,10 @@ public DocumentServiceLeaseUpdaterImpl(ChangeFeedContextClient client) {
                         cachedLease.setConcurrencyToken(serverLease.getConcurrencyToken());
                         cachedLease.setOwner(serverLease.getOwner());
 
-                        throw new LeaseConflictException(cachedLease, "Partition update failed");
+                        return Mono.error(new LeaseConflictException(cachedLease, "Partition update failed"));
                     });
             })
-            .retry(RETRY_COUNT_ON_CONFLICT, throwable -> {
+            .retryWhen(Retry.max(RETRY_COUNT_ON_CONFLICT).filter(throwable -> {
```

Comment:
This change is required as `retry(int, Predicate<Throwable>)` was removed in Reactor 3.4.0

Reply:
Potentially look into Retry.withThrowable(RetryUtils.toRetryWhenFunc(retryPolicy))